geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/pubsub/AbstractSubCommandsIntegrationTest.java

```diff
@@ -41,24 +44,32 @@
 import org.apache.geode.redis.internal.netty.Coder;
 import org.apache.geode.redis.mocks.MockSubscriber;
 import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.junit.rules.ExecutorServiceRule;
 
 public abstract class AbstractSubCommandsIntegrationTest implements RedisIntegrationTest {
+
+  @ClassRule
+  public static ExecutorServiceRule executor = new ExecutorServiceRule();
+
   private Jedis subscriber;
   private Jedis introspector;
   private MockSubscriber mockSubscriber;
 
   @Before
   public void setup() {
     mockSubscriber = new MockSubscriber();
-    subscriber = new Jedis("localhost", getPort());
-    introspector = new Jedis("localhost", getPort());
+    subscriber = new Jedis(BIND_ADDRESS, getPort(), REDIS_CLIENT_TIMEOUT);
+    introspector = new Jedis(BIND_ADDRESS, getPort(), REDIS_CLIENT_TIMEOUT);
   }
 
   @After
   public void teardown() {
     if (mockSubscriber.getSubscribedChannels() > 0) {
       mockSubscriber.unsubscribe();
     }
+    if (mockSubscriber.getSubscribedChannels() > 0) {
+      mockSubscriber.punsubscribe();
+    }
```

Comment:
Can this be combined with the if statement above it?

Reply:
There is a race condition here. In the jedis code,  if the subscribers end up going to 0 then there is a possiblilty of a NPE for subsequent (p)unsubscribe calls if we don't first do the check.